watertap/property_models/ion_DSPMDE_prop_pack.py

```diff
@@ -0,0 +1,814 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+Initial property package for multi-ionic system for use in the
+Donnan Steric Pore Model with Dielectric Exclusion (DSPM-DE)
+"""
+
+# Import Python libraries
+import idaes.logger as idaeslog
+
+from enum import Enum, auto
+# Import Pyomo libraries
+from pyomo.environ import Constraint, Expression, Reals, NonNegativeReals, \
+    Var, Param, Suffix, value, check_optimal_termination, units as pyunits, assert_optimal_termination
+from pyomo.common.config import ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import (declare_process_block_class,
+                        MaterialFlowBasis,
+                        PhysicalParameterBlock,
+                        StateBlockData,
+                        StateBlock,
+                        MaterialBalanceType,
+                        EnergyBalanceType)
+from idaes.core.components import Component, Solute, Solvent
+from idaes.core.phases import LiquidPhase
+from idaes.core.util.constants import Constants
+from idaes.core.util.initialization import (fix_state_vars,
+                                            revert_state_vars,
+                                            solve_indexed_blocks)
+from idaes.core.util.misc import add_object_reference, extract_data
+from idaes.core.util import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom, \
+    number_unfixed_variables
+from idaes.core.util.exceptions import ConfigurationError, InitializationError
+import idaes.core.util.scaling as iscale
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+class ActivityCoefficientModel(Enum):
+    ideal = auto()                    # Ideal
+    davies = auto()                   # Davies
+
+
+@declare_process_block_class("DSPMDEParameterBlock") #TODO: replace w/ "PropParameterBlock"
```

Comment:
```suggestion
@declare_process_block_class("PropParameterBlock")
```

Reply:
Do we really want to standardize these? Whilst it might sound like a good idea now, I can see it causing confusion later. Consider a case where I want to use two property packages in a flowsheet, and they both use standard names - now they need to be imported with aliases (so the user is effectively giving them unique names anyway).

Reply:
I had the same thought upon further reflection. Maybe it's best to keep this as is instead of trying to standardize.